### ISSUE TITLE
v8 - The database failed to upgrade, unknown state "<GUID>"

### ISSUE THREAD

**Issue by PerplexDaniel** [umbraco/Umbraco-CMS#4576]
When trying to upgrade I receive the following error message during the upgrade:

> The database failed to upgrade. ERROR: The database configuration failed with the following message: Unknown state "98339BEF-E4B2-48A8-B9D1-D173DC842BBE". Please check log file for additional information (can be found in '/App_Data/Logs/UmbracoTraceLog.txt')

`From: 8.0.0-alpha.58.2240`
`To: 8.0.0-alpha.58.2312`

Umbraco is unsable after this without reverting the whole upgrade.

I installed both from NuGet using the URL `https://www.myget.org/F/umbracocore/api/v3/index.json`, I did not have issues with other upgrades.

I am aware of the new database upgrade code which uses state transitions, so presumably the v2240 state is not registered in the code with an upgrade path. 

I realize this is still alpha, I am mainly reporting to prevent this from happening in production versions later, perhaps some sort of fallback and/or test case can be added for stu

### Bug summary

During upgrade from `8.0.0-alpha.58.2240` to `8.0.0-alpha.58.2312` the following exception is thrown and upgrade cannot continue. Umbraco cannot start after this:

> The database failed to upgrade. ERROR: The database configuration failed with the following message: Unknown state "98339BEF-E4B2-48A8-B9D1-D173DC842BBE". Please check log file for additional information (can be found in '/App_Data/Logs/UmbracoTraceLog.txt')

### Specifics

`From: 8.0.0-alpha.58.2240`
`To: 8.0.0-alpha.58.2312`

### Steps to reproduce

1. Install `8.0.0-alpha.58.2240` 
2. Install `8.0.0-alpha.58.2312` 
3. Observe error

**Comment by zpqrtbnk** [comment]
Dammit - I thought it only happened on my machine. 

What happens is this: I added a migration to the plan as such:
```
To<AddUserLoginDtoDateIndex>("98339BEF-E4B2-48A8-B9D1-D173DC842BBE");
```
then a few minutes later, I realized the Guid was not formatted like the other, so I fixed as:
```
To<AddUserLoginDtoDateIndex>("{98339BEF-E4B2-48A8-B9D1-D173DC842BBE}");
```
But apparently... the bogus version got pushed and released at some point ;-(

In "real" life I would not have updated the Guid - just to avoid this situation - but I really thought I had *not* pushed yet.

Easy fix: update umbracoKeyValue table, add the missing curlies to the Guid.

Sorry about that!
